src/service/InMemoryHistoryManager.java

```diff
@@ -2,32 +2,110 @@
 
 import model.*;
 
-import java.util.ArrayList;
-import java.util.LinkedList;
-import java.util.List;
+import java.util.*;
 
 public class InMemoryHistoryManager implements HistoryManager {
-
-    private static final byte HISTORY_LIST_MAX_SIZE = 10;
-    private final LinkedList<Task> historyTasks; // История изменений
+    private final CustomDoublyLinkedList<Task> historyTasks; // История изменений
+    private final Map<Integer, Node<Task>> linkedHashList;
 
     public InMemoryHistoryManager() {
-        this.historyTasks = new LinkedList<>();
+        this.historyTasks = new CustomDoublyLinkedList<Task>();
```

Comment:
Дублировать `Тип` при присвоении не надо, компилятор уже знает, какого типа объект
```suggestion
        this.historyTasks = new CustomDoublyLinkedList<>();
```